include/flux/op/find.hpp

```diff
@@ -6,44 +6,71 @@
 #ifndef FLUX_OP_FIND_HPP_INCLUDED
 #define FLUX_OP_FIND_HPP_INCLUDED
 
+#include <flux/core/utils.hpp>
 #include <flux/op/for_each_while.hpp>
 
+#include <cstring>
+#include <type_traits>
+
 namespace flux {
 
 namespace detail {
 
 struct find_fn {
+private:
+    template <sequence Seq, typename Value>
+    constexpr auto impl(Seq&& seq, Value const& value) const -> cursor_t<Seq>
+    {
+        return for_each_while(
+            seq, [&](auto&& elem) { return FLUX_FWD(elem) != value; });
+    }
+
+public:
     template <sequence Seq, typename Value>
         requires std::equality_comparable_with<element_t<Seq>, Value const&>
-    constexpr auto operator()(Seq&& seq, Value const& value) const -> cursor_t<Seq>
+    constexpr auto operator()(Seq&& seq, Value const& value) const
+        -> cursor_t<Seq>
     {
-        return for_each_while(seq, [&](auto&& elem) {
-            return FLUX_FWD(elem) != value;
-        });
+        constexpr auto memchr_opt = flux::contiguous_sequence<Seq> &&
+            flux::sized_sequence<Seq> && std::is_integral_v<Value> &&
+            flux::detail::any_of<value_t<Seq>, char, signed char, unsigned char,
+                const char, const signed char, const unsigned char>;
```

Comment:
This should be `can_memchr` for consistency with the other functions. 

I'm not quite sure that I understand the MSVC condition here allowing `Value` to be any integral type -- don't we need to also have `sizeof(Value) == 1` since `memchr` looks for a single byte? But maybe I'm missing something?

Conservatively (at least until I get my head around the above!) I think we should require that `Value` is the same type as `value_t<Seq>`, which is one of the "known good" types -- but i think we can also expand that list to include C++20 `char8_t` and `std::byte`. (We don't need to list the const variants as `value_t` shouldn't be const-qualified.)

```suggestion
        constexpr auto can_memchr = 
            contiguous_sequence<Seq> && sized_sequence<Seq> && 
            std::same_as<Value, value_t<Seq>> &&
            detail::any_of<value_t<Seq>, char, signed char, unsigned char, char8_t, std::byte>;
```

Reply:
> I'm not quite sure that I understand the MSVC condition here allowing Value to be any integral type

I did this because I saw that the the search value type in the function signature of `std::memchr` is `int` but I realize now it's meant to be a byte. 🤦‍♂️